### DISCUSSION TITLE
`NOW()` in `.from().select().lte()`

### DISCUSSION THREAD

**Discussion by jcuenod** [supabase/supabase#10853] [Questions] [answered]
I want to check whether a user's subscription has expired. I planned to write:

```js
await supabase
  .from("subscriptions")
  .select("*")
  .lte("NOW()::date", "subscription_expiry::date")
```

But this gives me:

> column subscriptions.NOW does not exist

**Comment by GaryAustin1** [comment]
Your first item in .lte is the column name, the 2nd is the value.  You can't use now().

You can use javascript time though like here: https://github.com/supabase/supabase/discussions/2351

If you have to use server time then you will need to do an rpc call.

**Comment by steve-chavez** [comment, marked as answer]
You can use `now` as a value, not as a column as mentioned by Gary. Try doing it backwards:

```js
await supabase
  .from("subscriptions")
  .select("*")
  .gt("subscription_expiry", "now")
```

**Reply by dshukertjr** [reply]
Wow, I did not know that this is possible!

**Reply by dshukertjr** [reply]
@steve-chavez Two questions:

1. I know that we can use `count` like `.select('*, foreign_table(count)')`, but are there other keywords like `count` or `now` that might be handy? Are there any docs that I can refer to?
2. Would you mind if I turn this into a [#SupabaseTips tweet](https://twitter.com/search?q=%23supabasetips&src=typed_query&f=live)?

**Reply by GaryAustin1** [reply]
I did not know it was possible either. Can’t find it documented in Postgrest.org.  At least searching on now.

**Reply by steve-chavez** [reply]
@dshukertjr @GaryAustin1 It is somewhat documented on https://postgrest.org/en/stable/how-tos/working-with-postgresql-data-types.html#timestamps. Though the special values need to be more visible.

> I know that we can use count like .select('*, foreign_table(count)'), but are there other keywords like count or now that might be handy?

Besides  `now`, you can use `tomorrow`, `yesterday` and all of the values stated on https://www.postgresql.org/docs/current/datatype-datetime.html#DATATYPE-DATETIME-SPECIAL-TABLE.

That works because on SQL those are values as well:

```sql
postgres=# select 'yesterday'::date;
    date
------------
 2022-12-11
(1 row)

postgres=# select 'tomorrow'::timestamp;
      timestamp
---------------------
 2022-12-13 00:00:00
(1 row)
```

> Would you mind if I turn this into a [#SupabaseTips tweet](https://twitter.com/search?q=%23supabasetips&src=typed_query&f=live)?

Sure! Go ahead!
